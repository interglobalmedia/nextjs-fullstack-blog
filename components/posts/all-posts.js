import PostsGrid from './posts-grid'
import classes from '../../styles/all-posts.module.scss'

function AllPosts(props) {
    const { posts } = props

    return (
        <section className={classes.posts}>
            <h1>All Posts</h1>
            <PostsGrid posts={posts} />
        </section>
    )
}

export default AllPosts