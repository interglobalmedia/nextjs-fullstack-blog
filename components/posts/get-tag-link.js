import Link from 'next/link'
import classes from '../../styles/post-item.module.scss'

export const getTagLink = (tag) => {
    return (
        <Link className={`${classes.tag} tag`} href={`/blog/tag/${tag}`}>#{tag}</Link>
    )
}