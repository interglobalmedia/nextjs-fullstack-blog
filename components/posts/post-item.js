import Link from 'next/link'
import Image from 'next/image'
import classes from './post-item.module.scss'

function PostItem(props) {

    const { post } = props

    const options = {
        year: "numeric",
        month: "long",
        day: "numeric",
    };

    const formattedDate = new Date(post.date).toLocaleDateString('en-US', options)

    console.log(formattedDate)

    const imagePath = `/images/posts/${post.slug}/${post.image}`

    const linkPath = `/posts/${post.slug}`
    return (
        <li className={classes.post}>
            <Link href={linkPath}>
                <figure className={classes.image}>
                    <Image src={imagePath} alt={post.title} width={300} height={200} layout="responsive" legacyBehavior />
                    <figcaption className={classes.content}>
                        <h3>{post.title}</h3>
                        <time>{formattedDate}</time>
                        <p>{post.excerpt}</p>
                    </figcaption>
                </figure>
            </Link>
        </li>
    )
}

export default PostItem