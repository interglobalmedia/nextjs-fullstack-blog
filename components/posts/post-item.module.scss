.post {
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.2);
  background-color: var(--color-grey-800);
  text-align: center;
}

.post a {
  color: var(--color-grey-100);
}

.image {
  width: 100%;
  max-width: 800px;
  // max-height: 25rem;
  /* add so that things would display correctly and figcaption would display at all because set to position: absolute. */
  position: relative;
  // overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0 auto;
  background-color: var(--color-grey-800);
  // grid-column-start: auto;
  // grid-row-start: auto;
  color: #fff;
  background-size: cover;
  background-position: center;
  box-shadow: -2px 2px 10px 0 rgb(68 68 68 / 40%);
  transition: transform 0.3s ease-in-out;
  cursor: pointer;
  // counter-increment: item-counter;
}

.img {
  display: block;
  margin: 0 auto;
  height: 100%;
  width: 100%;
  max-width: 800px;
}

.image>img {
  object-fit: cover;
  height: 100%;
  width: 100%;
}

.image::after {
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: #000;
  opacity: 0.3;
  transition: opacity 0.3s ease-in-out;
}

.image:hover {
  transform: scale(1.05);
  background-color: var(--color-grey-800);
}

.image:hover::after {
  opacity: 0;
  background-color: var(--color-grey-800);
}

.content {
  position: relative;
  z-index: 1;
  color: #828282;
  // background: #eee;
  text-transform: lowercase;
  letter-spacing: 1px;
  /* This is to make sure that the figcaption is spaced the same in terms of parts, i.e. title, date, author, and then excerpt. */
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.content::after {
  // content: counter(item-counter);
  font-weight: bold;
  font-size: 1.1rem;
  padding-right: 0.5em;
  padding-left: 1rem;
  color: #444;
}

.content {
  padding: var(--size-4);
}

.content h3 {
  margin: var(--size-2) 0;
  font-size: var(--size-6);
  color: var(--color-grey-100);
}

.content time {
  font-style: italic;
  color: var(--color-grey-400);
}

.content p {
  line-height: var(--size-6);
  color: var(--color-grey-200);
}