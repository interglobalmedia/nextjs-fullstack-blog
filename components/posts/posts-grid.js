
import dynamic from 'next/dynamic'
import { useState } from 'react'
import classes from '../../styles/posts-grid.module.scss'

const DynamicPostItem = dynamic(() => import('./post-item'))

function PostsGrid(props) {
    const { posts, initialDisplayPosts = [] } = props
    const [searchValue, setSearchValue] = useState('')
    const filteredPosts = posts.filter((frontMatter) => {
        const searchContent =
            frontMatter.excerpt +
            frontMatter.tags.join(' ')
        return searchContent.toLowerCase().includes(searchValue.toLowerCase())
    })

    /* If initialDisplayPosts exist, display it if no searchValue is specified */
    const displayPosts =
        initialDisplayPosts.length > 0 && !searchValue
            ? initialDisplayPosts
            : filteredPosts

    return (
        <>
            <div className="relative">
                <input
                    aria-label="Search articles"
                    type="text"
                    onChange={(e) => setSearchValue(e.target.value)}
                    placeholder="Search articles"
                    className={`${classes.search} block w-full rounded-md bg-primary-500 border border-gray-300 px-4 py-2 text-gray-900 focus:border-primary-500 focus:ring-primary-500 dark:border-gray-900 dark:bg-gray-800 dark:text-gray-100`}
                    autoFocus
                />
                <svg
                    className="absolute right-3 top-3 h-5 w-5 text-gray-400 dark:text-gray-300"
                    xmlns="http://www.w3.org/2000/svg"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                >
                    <path
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth={2}
                        d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    />
                </svg>
            </div>
            <ul
                className={`grid ${classes.grid}`}
            >
                {!filteredPosts.length && 'No posts found.'}
                {
                    displayPosts.map(post => (
                        <DynamicPostItem key={post.slug} post={post} />
                    ))
                }
            </ul>
        </>
    )
}

export default PostsGrid