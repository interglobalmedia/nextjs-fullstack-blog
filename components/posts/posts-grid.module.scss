.grid {
  list-style: none;
  margin: 0;
  padding: 0;
  display: grid;
  align-items: stretch;
  margin: 3rem auto 6rem;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(20rem, 1fr));
  /* Added this so that there is a gap between the rows as well */
  gap: 1.5rem;
  align-content: center;

}