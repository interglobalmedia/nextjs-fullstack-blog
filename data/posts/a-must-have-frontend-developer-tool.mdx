---
slug: a-must-have-frontend-developer-tool
date: '2017-02-02T21:00:54.362Z'
lastModified: '2023-02-08T15:40:54.362Z'
title: A Must-Have Frontend Developer Tool
author: Maria D. Campbell
excerpt: A must-have frontend developer tool.
image: how-big-is-my-browser.jpg
tags:
    [
        'browser window size',
        'chrome extensions',
        'frontend development',
        'image size matching',
        'tools',
        'time savers',
        'workflows'
    ]
isPublished: true
isFeatured: false
---

![Screenshot of How Big Is My Browser tool](how-big-is-my-browser.jpg)

*Screenshot by
[Maria D. Campbell](https://www.interglobalmedianetwork.com/about) on
[Maria's Next Blog](/images/posts/a-must-have-frontend-developer-tool/how-big-is-my-browser.jpg)*

***Today*** I was ***adding*** a ***new project*** `image` to ***my*** `front end developer portfolio`, and I ***had*** the ***damnedest*** time ***trying*** to ***match*** its `size` to ***all*** the ***others***, which I ***had created*** with the
[FireShot Screen Capture](https://chrome.google.com/webstore/detail/take-webpage-screenshots/mcbpblocgmgfnpjjppndjkmgjaogfceg) `Chrome` ***extension*** at ***another time***. The ***trick*** to ***making all*** your `screen captures`
the ***same size*** is in the `size` of the `browser window`. But ***how*** was I ***going*** to figure ***that out***? I was ***getting*** `sick` and `tired` of ***redoing*** all `screen captures` ***just because*** they ***weren’t*** all ***taken*** at the ***same time***, ***with*** the ***probability*** that
the `images` ***would*** end up ***being*** `different sizes`. An ***experienced*** `Frontend Developer` ***knows*** that ***all*** `images` ***within*** a ***given*** `element` ***should be*** the ***same size*** in ***order*** for the `layout` to ***work***.

I ***took*** it ***upon myself*** to ***find*** a `tool` that would ***provide me*** with `browser window size`. I ***came up*** with ***just*** the ***right one*** in my `Google`` ***search***:

- [How Big Is My Browser?](http://howbigismybrowser.com/)

It ***does*** exactly ***what*** it ***says***. It ***tells*** you `how big` your `browser window` ***is***. What’s ***even better***, is that you can ***resize*** it if the `size` ***changes*** on you! ***No more*** `guessing games`. ***No more*** `re-dos`. ***No more*** `time wasting`. ***Check*** it ***out***!

*Happy* `Frontend` *developing*!
