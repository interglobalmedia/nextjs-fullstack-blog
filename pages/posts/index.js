import AllPosts from '../../components/posts/all-posts'

const DUMMY_POSTS = [
  {
    slug: 'getting-started-with-nextjs',
    title: 'Getting Started with Nextjs',
    image: 'getting-started-nextjs.png',
    excerpt: 'Nextjs is the React framework for production. It makes building fullstack React apps and sites a breeze, and ships with built-in SSR.',
    date: '2023-01-17'
  },
  {
    slug: 'getting-started-with-nextjs2',
    title: 'Getting Started with Nextjs',
    image: 'getting-started-nextjs.png',
    excerpt: 'Nextjs is the React framework for production. It makes building fullstack React apps and sites a breeze, and ships with built-in SSR.',
    date: '2023-01-17'
  },
  {
    slug: 'getting-started-with-nextjs3',
    title: 'Getting Started with Nextjs',
    image: 'getting-started-nextjs.png',
    excerpt: 'Nextjs is the React framework for production. It makes building fullstack React apps and sites a breeze, and ships with built-in SSR.',
    date: '2023-01-17'
  },
  {
    slug: 'getting-started-with-nextjs4',
    title: 'Getting Started with Nextjs',
    image: 'getting-started-nextjs.png',
    excerpt: 'Nextjs is the React framework for production. It makes building fullstack React apps and sites a breeze, and ships with built-in SSR.',
    date: '2023-01-17'
  }
]

function AllPostsPage() {
    return (
        <AllPosts posts={DUMMY_POSTS} />
    )
}

export default AllPostsPage