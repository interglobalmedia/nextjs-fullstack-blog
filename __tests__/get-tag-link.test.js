import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Link from 'next/link'
import { getTagLink } from '../components/posts/get-tag-link'

describe('getTagLink', () => {
	it('renders a tag link', () => {
		const tagMacOS = getTagLink(`macOS`)
		const tagLink = <Link href={tagMacOS}>#{tagMacOS}</Link>
		const tree = renderer.create(tagLink).toJSON()

		expect(tree).toMatchSnapshot()
	})
})
