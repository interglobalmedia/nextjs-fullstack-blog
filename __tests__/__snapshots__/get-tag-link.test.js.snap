// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTagLink renders a tag link 1`] = `
<a
  href=""
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  #
  <a
    className="tag tag"
    href="/blog/tag/macOS"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    #
    macOS
  </a>
</a>
`;
